internal/controller/dpuoperatorconfig_controller.go

```diff
@@ -97,8 +100,12 @@ func (r *DpuOperatorConfigReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		logger.Error(err, "Failed to get DpuOperatorConfig resource")
 		return ctrl.Result{}, err
 	}
-
-	err := r.ensureDpuDeamonSet(ctx, dpuOperatorConfig)
+	err := r.ensureVSP(ctx, dpuOperatorConfig)
+	if err != nil {
+		logger.Error(err, "Failed to ensure VSP is running")
+		return ctrl.Result{}, err
+	}
+	err = r.ensureDpuDeamonSet(ctx, dpuOperatorConfig)
```

Comment:
Please take a look at the failing test: https://github.com/openshift/dpu-operator/blob/main/internal/controller/dpuoperatorconfig_controller_test.go#L223

With this change the dpu daemon is not getting started properly, possibly due to an earlier failure in the controller when starting the vsp.

Reply:
I think platform is IPU.

This code expects vspDaemonImage = intel_VSP_IMAGE .
r.vspDaemonImage = os.Getenv(vendorName + "_" + "VSP_IMAGE")

In marvell case, it is like this:
config/dev/local-images-template.yaml:
+        - name: marvell_VSP_IMAGE
+          value: {{ .RegistryURL }}:5000/mrvl-vsp:dev

Is it the cause?

Reply:
I suspect ensureVSP() is failing due to lack of env intel_VSP_IMAGE:
https://github.com/openshift/dpu-operator/pull/82/commits/ebbe5ee264b0f013531e41f05304c6de4b5d0e9c#diff-ac97373df236b7f27a14df2d71cf3b436098b60ff6ae8fccdb17c6a6266a33ecR204

if ensureVSP() fails, _err = r.ensureDpuDeamonSet(ctx, dpuOperatorConfig)_ won't be called.

We changed our code to log error instead of return error. 

This might help keep it running on IPU.

Reply:
Yes currently make-test does not test for the vsp deployment, this is still outside the scope of the dpu operator. It is just validating that the daemon is properly created when the operator config is added.